portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,129 @@
+  const commentDate = document.createElement('p');
+  commentDate.innerText = humanReadableDateFromTimestamp(comment.timestamp);
+  // header for both username and date
+  const commentHeader = document.createElement('div');
+  commentHeader.classList.add('comment-header');
+  commentHeader.appendChild(commentAuthorUsername);
+  commentHeader.appendChild(commentDate);
+  return commentHeader;
+}
+
+function getCommentReplyForm(comment) {
+  // next I need a form for posting a reply to this comment
+  const replyForm = document.createElement('form');
+  replyForm.action = '/comments';
+  replyForm.method = 'POST';
+  const formReplyUsername = document.createElement('input');
+  formReplyUsername.type = 'text';
+  formReplyUsername.name = 'username';
+  formReplyUsername.placeholder = 'Your username';
+  const formReplyText = document.createElement('input');
+  formReplyText.type = 'text';
+  formReplyText.name = 'text';
+  formReplyText.placeholder = 'Write reply here';
+  // this element is invisible but it holds the replyTo id
+  const formReplyToId = document.createElement('input');
+  formReplyToId.type = 'text';
+  formReplyToId.name = 'replyTo';
+  formReplyToId.value = comment.key;
+  formReplyToId.hidden = true;
+  const formReplySubmit = document.createElement('input');
+  formReplySubmit.type = 'submit';
+  // now add all this form elements to the form
+  replyForm.appendChild(formReplyUsername);
+  replyForm.appendChild(formReplyText);
+  replyForm.appendChild(formReplyToId);
+  replyForm.appendChild(formReplySubmit);
+  replyForm.classList.add('hidden');
+  replyForm.id = comment.key + '_form';
+  return replyForm;
+}
+// toggles visibility of the reply form
+function toggleReplyField(commentId) {
+  const showReplyFormButtonId = commentId + '_button'; // generate button's id
+  const replyFormId = commentId + '_form';             // generate reply form id
+  const button = document.getElementById(showReplyFormButtonId);
+  button.innerText = (button.innerText === 'Cancel' ? 'Reply' : 'Cancel'); // change text depending on the mode
+  const replyForm = document.getElementById(replyFormId);
+  if (replyForm.classList.contains('hidden')) {
+    // if the form is hidden, show it
+    replyForm.classList.remove('hidden');
+    replyForm.classList.add('shown');
+  } else {
+    // otherwis hide the form
+    replyForm.classList.remove('shown');
+    replyForm.classList.add('hidden');
+  }
+}
+
+function humanReadableDateFromTimestamp(timestamp) {
+  const dateObj = new Date(Number(timestamp));
+  // 10 first characters (yyyy-mm-dd) are human readable, other is not
+  return dateObj.toISOString().slice(0, 10) + ' at ' +
+         (dateObj.getHours() < 10 ? '0' + dateObj.getHours() : dateObj.getHours()) + ':' +
+         (dateObj.getMinutes() < 10 ? '0' + dateObj.getMinutes() : dateObj.getMinutes()) + ':' +
+         (dateObj.getSeconds() < 10 ? '0' + dateObj.getSeconds() : dateObj.getSeconds());
+}
```

Comment:
Can we try including any library (e.g. jQuery) if that helps? Pretty sure they'll provide some utility formatters to do this in an easier way.

Otherwise, this should be at least separated into a 'util.js' file.